src/pages/Environments.js

```diff
@@ -132,82 +179,173 @@ const Environments = () => {
     size: 'size'
   }[diskSort.field]], [diskSort.direction], disks)
 
-  const totalCost = _.sum(_.map(runtimeCost, runtimes))
+  const filteredApps = _.orderBy([{
+    project: 'googleProject',
+    status: 'status',
+    created: 'auditInfo.createdDate',
+    accessed: 'auditInfo.dateAccessed',
+    cost: getGalaxyCost
+  }[appSort.field]], [appSort.direction], apps)
+
+  const filteredCloudEnvironments = _.concat(filteredRuntimes, filteredApps)
+
+  const totalRuntimeCost = _.sum(_.map(runtimeCost, runtimes))
+  const totalAppCost = _.sum(_.map(getGalaxyComputeCost, apps))
+  const totalCost = totalRuntimeCost + totalAppCost
   const totalDiskCost = _.sum(_.map(persistentDiskCostMonthly, disks))
 
   const runtimesByProject = _.groupBy('googleProject', runtimes)
   const disksByProject = _.groupBy('googleProject', disks)
+  const appsByProject = _.groupBy('googleProject', apps)
+
+  const renderBillingProjectApp = app => {
+    const inactive = !_.includes(app.status, ['DELETING', 'ERROR']) &&
+      currentApp(appsByProject[app.googleProject]) !== app
+    return h(Fragment, [
+      app.googleProject,
+      inactive && h(TooltipTrigger, {
+        content: 'This billing project has multiple active cloud environments. Only the most recently created one will be used.'
+      }, [icon('warning-standard', { style: { marginLeft: '0.25rem', color: colors.warning() } })])
```

Comment:
This tooltip showed up when I pressed the trash icon for an app on `Cloud Environments` page and it was briefly in `PREDELETING` status. Should we add that status to the list `inactive` statuses?

Also thinking more generally, does this tooltip still apply the way it was conceived since users can now have multiple cloud environments routinely and it may not make sense to notify them if they have multiple cloud environments because one is for Galaxy and one is a runtime?

Reply:
Also thinking more generally, does this tooltip still apply the way it was conceived since users can now have multiple cloud environments routinely and it may not make sense to notify them if they have multiple cloud environments because one is for Galaxy and one is a runtime?

Reply:
the warning triangle will not show up if the user has a runtime and galaxy app in the same billing project. the logic that handles this is in `renderBillingProjectApp` and `renderBillingProjectRuntime` - these methods ensure that the warning triable only shows up if there is more than one galaxy app in a project or more than one runtime in a project but i tested to make sure that it doesn't show up if the user has 1 galaxy app and 1 runtime in a project